Algorithm/QCAlgorithm.Trading.cs

```diff
@@ -957,48 +957,40 @@ public decimal CalculateOrderQuantity(Symbol symbol, decimal target)
             decimal marginRequired;
             decimal orderValue;
             decimal orderFees;
-            var feeToPriceRatio = 0;
+            var feeToPriceRatio = 0m;
 
             // compute the initial order quantity
-            decimal orderQuantity = targetOrderValue / unitPrice;
+            var orderQuantity = targetOrderValue / unitPrice;
 
-            if (orderQuantity % security.SymbolProperties.LotSize != 0)
-            {
-                orderQuantity = orderQuantity - (orderQuantity % security.SymbolProperties.LotSize);
-            }
-
-            var iterations = 0;
+            // rounding off Order Quantity to the nearest multiple of Lot Size
+            orderQuantity -= orderQuantity % security.SymbolProperties.LotSize;
```

Comment:
If `security.SymbolProperties.LotSize` is zero we would get a `DivideByZeroException` here,
maybe better add a check and throw an error message if zero ?

Reply:
If `security.SymbolProperties.LotSize` is non-positive, we should also throw and error message, because negative number is also incorrect, right? It would be an error in the `symbol-properties-database`
```
// if security lot size is non-positive, throw exception
if (security.SymbolProperties.LotSize <= 0)
{
    throw new Exception($"{security.Symbol} lot size cannot be less than or equal to zero");
}
```

Reply:
Maybe we should enforce these constraints in the symbol properties class instead of everywhere we consume it. I think enforcing a positive valued lot size is reasonable and correct.

Reply:
What value would be correct? It is normally 1, but FXCM Forex is 1000 and crypto is 0.00000001. We could throw in `SymbolPropertiesDatabase.FromCsvLine` where we read the data.

Reply:
All out of scope for this PR. Agree safe assumption the input data is correct at this level of the application. If it was incorrect it should through on load as lot size of 0 isn't realistic.